Makefile.pdlibbuilder

```diff
@@ -522,8 +523,33 @@ ifeq ($(system), Darwin)
     cxx.flags := -fcheck-new
   endif
   ifeq ($(extension), d_fat)
-    arch := i386 x86_64
+    ifeq ($(origin arch), undefined)
+      # detect macOS version from Xcode toolchain version & deduce archs
+      xcode.ver := $(shell \
+        pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | \
+        grep 'version' | sed 's/[^0-9]*\([0-9]*\).*/\1/')
+      ifeq ($(xcode.ver), undefined)
+        $(warn Could not find Xcode version. Using default d_fat archs.)
+        arch := i386 x86_64
+      else ifeq ($(shell [ $(xcode.ver) -gt 11 ] && echo 1), 1)
+        # Xcode 12+
+        arch := x86_64 arm64
+      else ifeq ($(shell [ $(xcode.ver) -gt 9 ] && echo 1), 1)
+        # Xcode 10 - 11
+        $(error Cannot build d_fat. Xcode version $(xcode.ver) \
+                only allows building for $(target.arch).)
```

Comment:
i wonder whether we really want an **error** here.

if the user specifies `d_fat` they obviously want *fat* binaries, but since they are didn't specify which architectures they want, they are only asking for "as fat as possible".
a fat binary with a single architecture is an edge case, but technically it still is a fat binary (and Pd will load it anyhow).

Reply:
>"as fat as possible"

Hah, yeah ok. I was being conservative as I've found it better to manage expectations and warn people, rather then expect them to know all of these details.

What do you think about allowing but printing a warning?

Reply:
Changed to a warning.